src/biogeophys/CanopyFluxesMod.F90

```diff
@@ -865,13 +884,53 @@ subroutine CanopyFluxes(bounds,  num_exposedvegp, filter_exposedvegp,
       do f = 1, fn
          p = filterp(f)
          c = patch%column(p)
+         g = patch%gridcell(p)
+
+         select case (z0param_method)
+         case ('ZengWang2007')
+            lt = min(elai(p)+esai(p), tlsai_crit)
+            egvf =(1._r8 - alpha_aero * exp(-lt)) / (1._r8 - alpha_aero * exp(-tlsai_crit))
+            displa(p) = egvf * displa(p)
+            z0mv(p)   = exp(egvf * log(z0mv(p)) + (1._r8 - egvf) * log(z0mg(c)))
+
+         case ('Meier2022')
+            lt = max(0.00001_r8,elai(p)+esai(p))
+            displa(p) = htop(p) * (1._r8 - (1._r8 - exp(-(cd1_param * lt)**0.5_r8)) / (cd1_param*lt)**0.5_r8)
+
+            lt = min(lt,z0v_LAImax(patch%itype(p)))
+            delt = 2._r8
+            U_ustar_ini = (z0v_Cs(patch%itype(p)) + z0v_Cr(patch%itype(p)) * lt / 2._r8)**(-0.5_r8) &
+                      *z0v_c(patch%itype(p)) * lt / 4._r8
```

Comment:
Found more divisions that could be * 0.5 and * 0.25 but will hold off for now...

Reply:
done